### ISSUE TITLE
[Feature] Tooltip support

### ISSUE THREAD

**Issue by hasufell** [daa84/neovim-gtk#113]
I'm not really sure what is involved in that feature. When using diagnostics via e.g. ALE, I get tooltips in gvim such as:

![peek 2018-06-15 20-20](https://user-images.githubusercontent.com/1241845/41483844-f576ee38-70da-11e8-8d5c-767a53d66fac.gif)

neovim-gtk currently doesn't seem to do that:

![peek 2018-06-15 20-29](https://user-images.githubusercontent.com/1241845/41483855-fe265bd6-70da-11e8-9d0c-901bde581e6a.gif)

**Comment by daa84** [comment]
I think this feature is disabled in neovim, but not sure. Just don't know where to get this tooltip info.

**Comment by hasufell** [comment]
> Afaik it's called balloons and it's removed in neovim indeed.

Then I wonder how this was implemented in [gonvim](https://github.com/dzhou121/gonvim):

![gonvim](https://user-images.githubusercontent.com/1241845/41498469-fccb6062-716e-11e8-982c-09f7ce881446.png)

**Comment by Anrock** [comment]
@hasufell they probably just reimplemented it on top of nvim somehow.

**Comment by daa84** [comment]
hm, yes it implemented using nvim api, https://github.com/dzhou121/gonvim/blob/c264bb83afbfe10c2d4991037142c169ce6519c3/editor/locpopup.go#L135

Interesting, think it is possible to implement same way

**Comment by hasufell** [comment]
> Interesting, think it is possible to implement same way

Right, that would be a shortcut, that will be useful short-term at least.

However, gvim seems to have a much more powerful approach there that even integrates with https://github.com/autozimu/LanguageClient-neovim showing type information on mouse-over:

https://i.imgur.com/C11BXgD.png

However, errors are the most important thing, probably.

**Comment by TerminalWitchcraft** [comment]
+1 for this feature. Tooltips are really helpful and it will be great if this is implemented.
